### DISCUSSION TITLE
Firefox 117 update

### DISCUSSION THREAD

**Discussion by lgander** [onemen/TabMixPlus#237] [Q&A] [answered]
I just updated firefox and TMP stopped working. 
Tried installing the last version and I got that it can't be installed because it appears to be corrupt.
Any help on this would be much appreciated.
Thanks!

**Comment by onemen** [comment, marked as answer]
![firefox](https://github.com/onemen/TabMixPlus/assets/3650909/e39c4d4e-5bec-47fe-96d7-faba7fab24b2)

1. Install latest scripts to your Application Binary folder, [config scripts](https://github.com/onemen/TabMixPlus/files/12257459/fx-folder.zip).
2. Profile Folder is the path to create chrome folder and extract [utils](https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip) folder in it.
3. Remove Tab mix extension.
4. go to `about:config` and set `extensions.experiments.enabled` to `true`
5. Open about:support and click "Clear startup cache…" to force Firefox to load userChromeJS on the next startup.
6. Start Firefox again.
7. Install Tab mix again.

**Reply by lgander** [reply]
That worked. Thanks a lot for the fast and detailed reply.

**Reply by heram01** [reply]
Hello,

So far I could keep the TMP working with ever changing FF Nightly, but not since this last update. FF had disabled the TMP and no way to re-enable it (yes, I have related settings overridden in About:config).
Please, may you provide a solution for Mac OS system? I don't understand how I can install latest "config scripts" to my Application Binary folder on Mac?
I don't understand where should I find the profile folder on Mac and how to install "utils" in it?
Thank you very much!

**Reply by heram01** [reply]
I was able to restore the FF to the previous version (116.0a1 (2023-06-21)) and now the latest TMP is back to work. But how can I upgrade to the current FF Nightly so the TMP would also work?

**Reply by heram01** [reply]
Any one out there to help?

**Reply by lgander** [reply]
I'm on windows so no clue about mac. But I bet that the correct path is detailed on the instructions. If it isn't, maybe start another thread specific to that may help?

**Reply by heram01** [reply]
what instructions? Where? My TMP self updated... then I permitted to FF Nightly to update itself. Then the FF had disabled the TMP. I've restored the previous version of FF from the Time Machine. So now the TMP is new, but the FF - previous (from June) version. And Nightly is annoying, every so often asks to be updated. Mac OS paths are quite different from those in Windows and I couldn't understand meaning of "install" - does it mean simply copy files or copy directory (where?) or something has to be done with them. Pain.

**Reply by lgander** [reply]
I dont use nightly neither, seems that involves an extra step.
But I guess you can get the paths for mac from this places and follow the same instructions that @onemen sent me on this post:
https://github.com/onemen/TabMixPlus#installation
https://github.com/xiaoxiaoflood/firefox-scripts#instructions

**Reply by heram01** [reply]
Thank you. Whatever I do - the TMP ends up being disabled after new (119) FF Nightly update. The only way to restore it back - it's to use Time Machine and roll back both, the FF and profiles to June's 116 version.

**Reply by onemen** [reply]
make sure to follow ALL the steps from [here](https://github.com/onemen/TabMixPlus/discussions/237#discussioncomment-6915374)

**Reply by heram01** [reply]
Thank you, I try to follow all these steps (while FF 116.03 works, it's 119 which disables the TMP).
But I do not understand this:
"Profile Folder is the path to create chrome folder and install [utils] folder in it".
First of all - what/where is that Profile Folder path in MacOS ? Which one? Second of all - what does it mean "install" [utils]? - is it just copy them? What where is the "chrome" folder? Does it have a name "chrome" - it doesn't exist on my Mac.
Please, may you just provide the paths or commands which would reveal those complete paths and directory (folder) name which should be created and whether the "util's" contect copied in it or the subdirectory "util" should copied in it.

It's really very hard every time restore older FF (116) and its profile while losing the current content of the day.

Thank you very much!

**Reply by onemen** [reply]
@heram01 

- open `about:support` and check what is the path for the current `Profile Folder`.
- in the profile folder create `chrome` folder (if one does not exist).
- in the new `chrome` folder you should extract the the content of https://github.com/onemen/TabMixPlus/files/12257460/utils_extensions_and_scripts.zip. the result should be single `utils` folder inside the `chrome` folder, the `utils` folder should contain all the scripts file not another folder.

don't use the same profile for Firefox 116 and Firefox nightly.

**Reply by heram01** [reply]
Thank you. 
These are profile directories. 
Default Profile	yes
Root Directory	/Users/grishab/Library/Application Support/Firefox/Profiles/p0af1nfj.default-1485044342641
Local Directory	/Users/grishab/Library/Caches/Firefox/Profiles/p0af1nfj.default-1485044342641

Did exactly as above. Created "chrome" and copied "utils" in it.
The result is also the same - the TMP is disabled as soon as updated  FF 119 started.
Also tried removing the TMP and reinstalling it again - the FF says that "add-on's" file is corrupted.

**Reply by onemen** [reply]
- make sure you install latest config.js file in the same folder that contains your Firefox binary file

- Remove Tab mix extension.
- close Firefox
- delete /Users/grishab/Library/Caches/Firefox/Profiles/p0af1nfj.default-1485044342641
- open Firefox
- install Tab mix extension.

**Reply by StavrosTsik** [reply]
@onemen 

Which configuration scripts are more up to date ? The scripts posted [here](https://github.com/onemen/TabMixPlus/discussions/237#discussioncomment-6915374) **or** [here](https://onemen.github.io/tabmixplus-docs/other/installation/) ? After a quick check, the files have different download locations and different Hash Values. :confused::confused:

**Reply by onemen** [reply]
https://onemen.github.io/tabmixplus-docs/other/installation/

**Comment by heram01** [comment]
Thank you. Followed exactly. Directory " /Users/grishab/Library/Caches/Firefox/Profiles/p0af1nfj.default-1485044342641" doesn't exist. The result - the same. When trying to install the TMP - it says that the file is corrupt and cannot be installed.

**Comment by heram01** [comment]
Why can't you register the TMP with Mozilla? - it looks like the only thing they want.

**Comment by onemen** [comment]
> Why can't you register the TMP with Mozilla? - it looks like the only thing they want.

it is impossible to register legacy extensions

**Comment by lonegungirl** [comment]
Tried reinstalling all the config/util files again but still no dice.  When I try to download the latest Tab Mix script, github just gives me a popup saying it won't install the previously downloaded extension because it's corrupt.

**Reply by onemen** [reply]
Right click on xpi file and use `save`

**Reply by lonegungirl** [reply]
Thanks!  I used "save link" because that was the only save option, but when I try to install it, dragging it into the extensions list doesn't seem to do anything, and when I try to install the add-on from a file, it tells me it is corrupt again.

**Reply by onemen** [reply]
Until you install config.js and utils scripts properly you will not be able to use `Tab Mix Plus`

**Reply by lonegungirl** [reply]
Tried doing everything over again and it worked this time!  I'm not sure if the fact that Firefox updated itself _again_ had anything to do with it, but not going to question the good fortune.  Thank you for all the assistance!
